controller/program/src/instructions/swap_exact_in.rs

```diff
@@ -0,0 +1,402 @@
+    program_error::{ProgramError, INVALID_ACCOUNT_DATA, NOT_ENOUGH_ACCOUNT_KEYS},
+};
+use sanctum_spl_token_jiminy::{
+    instructions::transfer::transfer_checked_ix_account_handle_perms,
+    sanctum_spl_token_core::{
+        instructions::transfer::{NewTransferCheckedIxAccsBuilder, TransferCheckedIxData},
+        state::{
+            account::{RawTokenAccount, TokenAccount},
+            mint::{Mint, RawMint},
+        },
+    },
+};
+
+use crate::{
+    instructions::sync_sol_value::sync_sol_val_with_retval,
+    pricing::NewPpIxPreAccsBuilder,
+    svc::NewSvcIxPreAccsBuilder,
+    verify::{verify_not_rebalancing_and_not_disabled, verify_pks},
+    Accounts, Cpi,
+};
+
+#[inline]
+pub fn process_swap_exact_in(
+    accounts: &mut Accounts<'_>,
+    args: &IxArgs,
+    cpi: &mut Cpi,
+) -> Result<(), ProgramError> {
+    if args.amount == 0 {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::ZeroValue).into());
+    }
+    let (ix_prefix, suf) = accounts
+        .as_slice()
+        .split_first_chunk()
+        .ok_or(NOT_ENOUGH_ACCOUNT_KEYS)?;
+
+    let ix_prefix = IxPreAccs(*ix_prefix);
+
+    let pool_state = *ix_prefix.pool_state();
+    let lst_state_list = *ix_prefix.lst_state_list();
+    let inp_lst_mint = *ix_prefix.inp_lst_mint();
+    let out_lst_mint = *ix_prefix.out_lst_mint();
+    let inp_pool_reserves = *ix_prefix.inp_pool_reserves();
+    let out_pool_reserves = *ix_prefix.out_pool_reserves();
+
+    let list = LstStatePackedList::of_acc_data(accounts.get(lst_state_list).data())
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstStateListData))?;
+
+    let inp_lst_state = list
+        .0
+        .get(args.inp_lst_index as usize)
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstIndex))?;
+
+    let inp_lst_state = unsafe { inp_lst_state.as_lst_state() };
+
+    let inp_lst_mint_acc = accounts.get(inp_lst_mint);
+    let inp_token_prog = inp_lst_mint_acc.owner();
+
+    let out_lst_state = list
+        .0
+        .get(args.out_lst_index as usize)
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstIndex))?;
+
+    let out_lst_state = unsafe { out_lst_state.as_lst_state() };
+
+    let out_lst_mint_acc = accounts.get(out_lst_mint);
+    let out_token_prog = out_lst_mint_acc.owner();
+
+    let expected_inp_reserves = create_raw_pool_reserves_addr(
+        inp_token_prog,
+        &inp_lst_state.mint,
+        &inp_lst_state.pool_reserves_bump,
+    )
+    .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidReserves))?;
+
+    let expected_out_reserves = create_raw_pool_reserves_addr(
+        out_token_prog,
+        &out_lst_state.mint,
+        &out_lst_state.pool_reserves_bump,
+    )
+    .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidReserves))?;
+
+    let expected_pks = NewSwapExactInIxPreAccsBuilder::start()
+        .with_lst_state_list(&LST_STATE_LIST_ID)
+        .with_pool_state(&POOL_STATE_ID)
+        .with_protocol_fee_accumulator(&PROTOCOL_FEE_ID)
```

Comment:
this aint the right account. This aint even a spl-token account - you can double check this on mainnet

https://github.com/igneous-labs/S/blob/master/docs/s-controller-program/accounts.md#protocol-fee-accumulators

Reply:
Does this look right? https://github.com/igneous-labs/inf-1.5/pull/77/commits/12a31ed12b54c22711a325694976c50e4381e087